usl_models/usl_models/flood_ml/model_params.py

```diff
@@ -0,0 +1,39 @@
+"""Flood model parameters."""
+
+from dataclasses import dataclass
+
+import tensorflow as tf
+
+from usl_models.flood_ml import constants
+
+
+@dataclass(kw_only=True)
```

Comment:
I think it's almost always a good idea to set `slots=True` for your dataclasses.

https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass

Reply:
Is this primarily for memory efficiency? I don't quite understand what this is doing.

Reply:
It's for memory efficiency, attribute access efficiently, and also avoids bugs when accidentally setting non-existent fields.

regular classes allow you to add new fields at runtime
```
>>> from dataclasses import dataclass
>>> @dataclass
... class Foo:
...   x: int
... 
>>> f = Foo(x=2)
>>> f.x
2
>>> f.y
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Foo' object has no attribute 'y'
>>> f.y = 4
>>> f.y
4
```

slots do not

```
>>> @dataclass(slots=True)
... class Foo:
...   x: int
... 
>>> f = Foo(x=2)
>>> f.x
2
>>> f.y = 4
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Foo' object has no attribute 'y'
```

which I'd say is generally a good thing!

Reply:
Thanks for the explanation! I wasn't aware of the extra protection against setting non-existent fields, very useful!